tests/main/econnreset/task.yaml

```diff
@@ -5,21 +5,25 @@ restore: |
 execute: |
     echo "Downloading a large snap in the background"
     su -c "/usr/bin/env SNAPD_DEBUG=1 snap download --edge test-snapd-huge 2>snap-download.log" test &
-    echo "Wait until it actually got some data for the snap"
-    while ! grep -q "Retrying.*download-snap/.*\.snap, attempt 1" snap-download.log; do
-        sleep 1
+
+    echo "Wait until the download started and downloaded more than 1 MB"
+    partial=test-snapd-huge_1.snap.partial
+    for i in $(seq 20); do
+        [ -f $partial ] && [ $(wc -c <"$partial") -gt 1048576 ] && break
+        sleep .5
     done
-    echo "Wait a little bit for the download to start, then force sending a RST package"
-    sleep 3
+    [ ! -f $partial ] || [ $(wc -c <"$partial") -eq 0 ] && exit 1
```

Comment:
(style nitpick): as above, I would prefer this slightly more explicit, something like
```
if [ ! -f "$partial" ] || [ $(wc -c <"$partial") -eq 0 ]; then
    echo "partial file $file did not start downloading, test broken"
   exit 1
fi
```
and same comment about `wc` vs `stat`. In this particular case you might consider using `[ ! -s "$partial" ]`  but then the two checks are fine with me, easier to read than to remember what `-s` means.